osidb/tests/endpoints/flaws/test_flaws.py

```diff
@@ -1819,3 +1820,66 @@ def test_flaw_history_no_history(self, auth_client, test_api_uri):
         body = response.json()
         flaw_json = body["results"][0]
         assert "pgh_diff" not in flaw_json["history"]
+
+    @pytest.mark.parametrize(
+        "workflow_status",
+        (
+            WorkflowModel.WorkflowState.NEW,
+            WorkflowModel.WorkflowState.REJECTED,
+            WorkflowModel.WorkflowState.DONE,
+        ),
+    )
+    @pytest.mark.parametrize(
+        "embargoed, internal", [(True, False), (False, False), (False, True)]
+    )
+    def test_flaw_available(
+        self,
+        workflow_status,
+        embargoed,
+        internal,
+        client,
+        test_api_uri,
+    ):
+        """
+        Test that API endpoint reports whether a flaw is available
+        based on the following criteria:
+
+        1) Public or work on flaw is done: 204 status
+        2) Not public and work not done / flaw does not exist: 404 status
+        3) CVE ID is not valid: 400 status
+        """
+
+        if internal:
+            flaw = FlawFactory(embargoed=False)
+            flaw.set_internal()
+        else:
+            # not embargoed is defaulted to public
+            flaw = FlawFactory(embargoed=embargoed)
+
+        AffectFactory(flaw=flaw)
+
+        flaw.workflow_state = workflow_status
+        assert flaw.save() is None
+
+        # response should return a state with no data
+        response = client.get(f"{test_api_uri}/available-flaws/{flaw.cve_id}")
+        assert response.data is None
+
+        if (
+            flaw.is_public
+            or flaw.classification["state"] == WorkflowModel.WorkflowState.REJECTED
+            or flaw.classification["state"] == WorkflowModel.WorkflowState.DONE
```

Comment:
You can access directly the workflow_state field instead of using the classification property.